core/utils.h

```diff
@@ -1946,12 +1946,12 @@ notify(syslog_event_type_t priority, bool internal, bool synch,
 #define SYSLOG_INTERNAL_CRITICAL_ONCE(...) \
     DODEBUG_ONCE(SYSLOG_INTERNAL_CRITICAL(__VA_ARGS__))
 
-#define MSG_COULDNT_SYNCH_WITH_ALL_THREADS_CODE 42
-
-#define fatal_error(id, sub, ...) \
+#define REPORT_FATAL_ERROR_AND_EXIT(dcontext, msg_id, arg_count, ...) \
     do { \
-      SYSLOG_COMMON(false, SYSLOG_CRITICAL, id, sub, __VA_ARGS__); \
-      exit_process_syscall(MSG_##id##_CODE); \
+        SYSLOG_COMMON(false, SYSLOG_CRITICAL, msg_id, arg_count, \
+                      __VA_ARGS__); \
+        os_terminate_with_code(dcontext, TERMINATE_PROCESS, 40); \
```

Comment:
Generally a named constant (#define or static const) is preferred to a hardcoded magic number literal: people reading this code are going to sit and puzzle over this "40" with no comment or name to indicate where it came from

Reply:
Good point, done.